test/Microsoft.AspNet.Mvc.IntegrationTests/BinderTypeBasedModelBinderIntegrationTest.cs

```diff
@@ -273,7 +264,22 @@ public Task<ModelBindingResult> BindModelAsync(ModelBindingContext bindingContex
 
                 var address = new Address() { Street = "SomeStreet" };
 
-                return Task.FromResult(new ModelBindingResult(address, bindingContext.ModelName, true));
+                bindingContext.ModelState.SetModelValue(
+                  bindingContext.ModelName + ".Street",
+                  new ValueProviderResult(
+                      address.Street,
+                      address.Street,
+                      System.Globalization.CultureInfo.CurrentCulture));
+
+                var validationNode = new ModelValidationNode(
+                  bindingContext.ModelName,
```

Comment:
Well this is the validation node for `Address` and not `Address.Street`, the validation node for `Address.Street` gets created automatically due to `ValidateAllProperties` flag set to true.